src/strategies/hedera/AccountServiceHedera.ts

```diff
@@ -279,6 +282,106 @@ export default class AccountServiceHedera implements IAccountService {
         return await this.apiService.getTransactionsFrom(accountAddress, transactionType, nextPage, transactionsLimit);
     }
 
+    async getAccountsFromMnemonic(
+        mnemonicRaw: string,
+        network: Network
+    ): Promise<AccountPrivateRecord[]> {
+        const mnemonic = await Mnemonic.fromString(
+            mnemonicRaw
+                .toLowerCase()
+                .trim()
+                .split(" ")
+                .filter((word) => word)
+                .join(" ")
+        );
+    
+        // derive to ECDSA Standard - find account
+        // derive to ECDSA Legacy - find account
+        // derive to ED25519 Standard - find account
+        // derive to ED25519 Legacy - find account
+        // return all records with account found. If no account show ECDSA Standard keys
+    
+        const records: AccountPrivateRecord[] = [];
+        let key: PrivateKey;
+        for (const keyType of Object.values(CryptoKeyType)) {
+            for (let standard = 1; standard >= 0; standard--) {
+                if (keyType === CryptoKeyType.ECDSA_SECP256K1) {
+                    if (standard) {
+                        key = await mnemonic.toStandardECDSAsecp256k1PrivateKey();
+                    } else {
+                        key = await mnemonic.toEcdsaPrivateKey();
+                    }
+                } else {
+                    if (standard) {
+                        key = await mnemonic.toStandardEd25519PrivateKey();
+                    } else {
+                        key = await mnemonic.toEd25519PrivateKey();
+                    }
+                }
+                records.push(...(await this.prepareAccountRecord(key, keyType, network)));
```

Comment:
may be improved with `Promise.all`, if you want to

Reply:
fixed ✅